app.go

```
@@ -0,0 +1,197 @@
+	dir string
+}
+
+func (f *filePersona) path() string {
+	return filepath.Join(f.dir, "SOUL.md")
+}
+
+func (f *filePersona) GetPersona(_ context.Context) (string, error) {
+	data, err := os.ReadFile(f.path())
+	if err != nil {
+		// Return default persona if file doesn't exist
+		return defaultPersona, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 174) but it returns nil

(nilerr)